kythe/cxx/extractor/cxx_extractor.cc

```diff
@@ -212,6 +212,13 @@ std::string FindStableRoot(
     const RepeatedPtrField<std::string>& arguments,
     const RepeatedPtrField<proto::CompilationUnit::FileInput>& required_input,
     const proto::CxxCompilationUnitDetails& details) {
+  for (const std::string& arg : arguments) {
+    if (absl::StartsWith(arg, "../")) {
+      LOG(WARNING) << "Using real working directory (" << working_directory
+                   << ") due to '..' prefix in argument: " << arg;
+      return std::string(working_directory);
+    }
+  }
```

Comment:
While this is likely sufficient, the working directory check below uses `find` to remain conservative and argument format agnostic.  It wants to find both plain `<value>` and `--foo=value` style arguments.  `../` is more likely to occur in "harmless" places within arguments, though, so perhaps changing this to `StartWith("../") or `Contains("=../")` would provide a bit more protection.

Additionally, I would check that none of the `required_input[].info.path` fields start with `../` as those files may not get added to the VFS in a reasonable location without a matching number of parent directories.

Reply:
Done, added a test for the new behavior as well.

Reply:
Thanks! Seems reasonable to me. @sam-mccall ?